### PR TITLE
`random` namespace refactoring

### PR DESCRIPTION
**Pull request by kelvinqian00**
- New `random` functions: `rng`, `rand-boolean`, and `choose-map`
- Rename `rand-long` to `rand-unbound-int`
- Rename `rand-gauss` to `rand-gaussian`
- Remove stars from function names (use `:refer-clojure :exclude` instead)
- Add function docstrings
- Remove useless computed mean from `choose`

### CODE REVIEW

File: src/main/com/yetanalytics/datasim/math/random.clj [hunk 1]
```diff
@@ -1,211 +1,238 @@
...
+  :ret ::rng)
+
+(defn rng
+  "Create a pseudorandom RNG using an arbitrary seed value."
+  ^Random []
+  (Random.))
 
 (s/fdef seed-rng
   :args (s/cat :seed ::seed)
   :ret ::rng)
```
Comment (milt) [inline src/main/com/yetanalytics/datasim/math/random.clj:45]: Is there a duped fdef here?

Comment (kelvinqian00) [reply src/main/com/yetanalytics/datasim/math/random.clj]: Not anymore: 07f303055955a6bc631b826c8acf8193a7cd1e0e

File: src/main/com/yetanalytics/datasim/math/random.clj [hunk 2]
```diff
@@ -1,211 +1,238 @@
...
-(s/fdef rand*
+;; Random Number Generation ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(s/fdef rand
   :args (s/cat :rng ::rng
                :n (s/? number?))
   :ret double?)
 
-(defn rand*
+(defn rand
```
Comment (milt) [inline src/main/com/yetanalytics/datasim/math/random.clj:60]: `rand*`, `rand-int*` etc were so named so they wouldn't collide with function names in core. Don't you get dreaded compilation warnings when they are named this way?

Comment (kelvinqian00) [reply src/main/com/yetanalytics/datasim/math/random.clj:60]: `:refer-clojure :exclude` was my friend

**Review by milt — APPROVED** [review]

**Pull request merged**